xknx/io/knxip_interface.py

```diff
@@ -64,8 +64,12 @@ def __init__(self,
         self.gateway_port = gateway_port
         self.auto_reconnect = auto_reconnect
         self.auto_reconnect_wait = auto_reconnect_wait
-        self.scan_filter = scan_filter
         self.bind_to_multicast_addr = bind_to_multicast_addr
+        if connection_type == ConnectionType.TUNNELING:
+            scan_filter.tunnelling = True
+        elif connection_type == ConnectionType.ROUTING:
+            scan_filter.routing = True
+        self.scan_filter = scan_filter
```

Comment:
GatewayScanFilter now accounting ConnectionType on initialization. I was thinking it wouldn't make sense to scan for routers if a tunneling connection shall be made (and vice versa).